sympy/solvers/solvers.py

```diff
@@ -912,7 +912,7 @@ def _sympified_list(w):
             f[i] = fi.as_expr()
 
         # rewrite hyperbolics in terms of exp
-        f[i] = f[i].replace(lambda w: isinstance(w, HyperbolicFunction),
+        f[i] = f[i].replace(lambda w: len(w.free_symbols & set(symbols)) and isinstance(w, HyperbolicFunction),
```

Comment:
I expect that it will be more efficient to reverse these checks e.g.:
```
isinstance(w, ...) and len(w.free_symbols ... )
```
Computing `free_symbols` requires recursing the expression tree where as `isinstance` is a one time check.

Reply:
I changed the order because the fact `1 and True` returns `True` but `True and 1` returns `1`.
```python
>>> print(1 and True)
True
>>> print(True and 1)
1
```
Although, I'll change the `len` expression to `len(w.free_symbols & set(symbols)) > 0` and then change the order. I hope that should work.

Reply:
It shouldn't be necessary to use `len` at all since an empty set is falsey.

Reply:
> It shouldn't be necessary to use `len` at all since an empty set is falsey.

Should I remove it?

Reply:
I don't think it really matters